### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by fabiaserra**
## Changes

* [Make sure version_id is a string so SG type set doesn't error out](https://github.com/ynput/ayon-shotgrid/commit/8fb62bfb52ed347c2fb3594d4c563fd73dbb10f2)

### CODE REVIEW

**Review by jakubjezek001 — CHANGES_REQUESTED** [review]
This would be good to merge once task issue is resolved

File: client/ayon_shotgrid/plugins/publish/collect_shotgrid_entities.py
Comment (fabiaserra) [reply client/ayon_shotgrid/plugins/publish/collect_shotgrid_entities.py:98]: if the task isn't collected here, the collect shotgrid entities completely fails... so no, it's not possible to collect it later or else we need to change the collect shotgrid entities plugin

**Comment by fabiaserra** [comment]
> This would be good to merge once task issue is resolved

what do you suggest doing?

**Comment by jakubjezek001** [comment]
> what do you suggest doing?

To address this, validation should occur later in the process, specifically during the validator stage, to prevent premature crashes.

**Comment by fabiaserra** [comment]
> To address this, validation should occur later in the process, specifically during the validator stage, to prevent premature crashes.

I'm still unsure what you are asking to do. The changes on this PR don't include any validation?

**Comment by BigRoy** [comment]
The 

> > To address this, validation should occur later in the process, specifically during the validator stage, to prevent premature crashes.
> 
> I'm still unsure what you are asking to do. The changes on this PR don't include any validation?

1. The change to `instance.data["task"]` **will** error due to a key error in certain instances/cases. That should be avoided.
If some other part absolute requires a task - then it should safely ignore that UNTIL the validation order so that a validator can turn it into a nice report.

2. Another approach, less pretty to the end user, is making this a dedicated `KnownPublishError` directly in the Collector - it will provide the artist with a not very informative UI (mostly saying "This is not your fault") but at the very least you're able to raise an error then for the logs that make it clearer as to why that edge case may occur.

1) is preferred because you can generate a helpful response to the artist as to why something is not working as intended.
But better yet, is to find a solution that doesn't require the task... because it's optional for instances, and preferably tools work with that if we can.

Does that help?

**Comment by BigRoy** [comment]
Thanks @fabiaserra 

I've removed this line from the PR description:

> * [Use list indexing instead of get as 'task' needs to always exist](https://github.com/ynput/ayon-shotgrid/commit/e9ff30c6eade9a4b20e9a1ba75db7018d5fadbfa)

---

@jakubjezek001 If there are issues with Shotgrid publishes when there is no explicit task set that'd now make that a separate issue to track and another PR. Making this PR ready for merging if you can test it.

**Comment by mkolar** [comment]
I'd be happy to merge too. Maybe @robin-ynput could have a look at testing this out?

**Comment by robin-ynput** [comment]
Tested locally and confirmed all works fine on my side. Was able to create both `PublishedFile`  and `Version` from Nuke properly.
@BigRoy @jakubjezek001 is that OK if I just go ahead and merge it ?

**Review by robin-ynput — APPROVED** [review]

**Comment by BigRoy** [comment]
I'd say, feel free to merge.

**Comment by robin-ynput** [comment]
@jakubjezek001 does it looks good to you ?
(I need your approval here since you've requested changes explicitly).

**Review by jakubjezek001 — APPROVED** [review]
yes, it is looking good. Thanks @fabiaserra

**Pull request merged**
